v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/database/CypherGenerator.java

```diff
@@ -295,30 +306,37 @@ private static Set<String> getRelationshipIndexAndConstraintsCypherStatements(Ta
             (mapping) -> isEntityProperty(mapping, FragmentType.rel) && mapping.isIndexed());
 
     String escapedType = ModelUtils.makeSpaceSafeValidNeo4jIdentifier(type);
-    for (String relKeyProperty : keyProperties) {
-      cyphers.add(
-          "CREATE CONSTRAINT IF NOT EXISTS FOR ()-[r:"
-              + escapedType
-              + "]-() REQUIRE r."
-              + ModelUtils.makeSpaceSafeValidNeo4jIdentifier(relKeyProperty)
-              + " IS RELATIONSHIP KEY");
-    }
-    for (String uniqueProperty : uniqueProperties) {
-      cyphers.add(
-          "CREATE CONSTRAINT IF NOT EXISTS FOR ()-[r:"
-              + escapedType
-              + "]-() REQUIRE r."
-              + ModelUtils.makeSpaceSafeValidNeo4jIdentifier(uniqueProperty)
-              + " IS UNIQUE");
+    if (capabilities.hasRelationshipKeyConstraints()) {
+      for (String relKeyProperty : keyProperties) {
+        cyphers.add(
+            "CREATE CONSTRAINT IF NOT EXISTS FOR ()-[r:"
+                + escapedType
+                + "]-() REQUIRE r."
+                + ModelUtils.makeSpaceSafeValidNeo4jIdentifier(relKeyProperty)
+                + " IS RELATIONSHIP KEY");
+      }
+
+      for (String uniqueProperty : uniqueProperties) {
```

Comment:
shouldn't this be outside the `if (capabilities.hasRelationshipKeyConstraints())` block?

Reply:
key constraints and unique constraints were introduced at the same time for relationships, so that's why I tied key constraints to unique constraints.

Reply:
done this refactor @fbiville.

Reply:
also added a couple of integration tests against different versions and editions.